js/plugins/mcp/README.md

```diff
@@ -13,53 +15,94 @@ To get started, you'll need Genkit and the MCP plugin:
 npm i genkit genkitx-mcp
 ```
 
-## MCP Client
+## MCP Client Manager
 
-To create an MCP client, you call the `mcpClient` function to generate a Genkit plugin for an MCP server. For example, to use MCP's example [filesystem server](https://github.com/modelcontextprotocol/servers/tree/main/src/filesystem):
+To connect to one or more MCP servers, you use the `createMcpClientManager` function. This function returns a Genkit plugin that manages connections to the configured MCP servers.
 
 ```ts
 import { genkit } from 'genkit';
-import { mcpClient } from 'genkitx-mcp';
+import { createMcpClientManager } from 'genkitx-mcp';
 
-// the filesystem server requires one or more allowed directories
+// Example: Configure a client manager for a local filesystem server
+// and a hypothetical remote Git server.
 const ALLOWED_DIRS = ['/Users/yourusername/Desktop'];
 
-const filesystemClient = mcpClient({
-  name: 'filesystem',
-  serverProcess: {
-    command: 'npx',
-    args: ['-y', '@modelcontextprotocol/server-everything', ...ALLOWED_DIRS],
+const mcpManager = createMcpClientManager({
+  name: 'myMcpClients', // A name for the manager plugin itself
+  mcpClients: {
+    // Each key (e.g., 'fs', 'git') becomes a namespace for the server's tools.
+    fs: { // Configuration for the first MCP server
+      serverProcess: {
+        command: 'npx',
+        args: ['-y', '@modelcontextprotocol/server-everything', ...ALLOWED_DIRS],
+      },
+    },
+    git: { // Configuration for a second MCP server (hypothetical)
+      serverUrl: 'http://localhost:8080/mcp',
+      // disabled: true, // Optionally disable a server
+    },
   },
 });
 
 const ai = genkit({
   plugins: [
-    filesystemClient /* ... other plugins such as model providers ...*/,
+    /* ... install plugins such as model providers ...*/
   ],
 });
+
+const mcpTools = await mcpManager.getAllTools(ai);
+
+// Provide MCP tools to the model of your choice.
+const response = await ai.generate({
+  model: gemini15Flash,
+  prompt: 'What are the last 5 commits in the repo `/home/test-repo/?`',
+  tools,
+});
 ```
 
+The `createMcpClientManager` function initializes a `GenkitMcpClientManager` instance, which handles the lifecycle and communication with the defined MCP servers.
+
+### `createMcpClientManager()` Options
+
+-   **`name`**: (required, string) A name for the client manager plugin itself.
+-   **`version`**: (optional, string) The version of the client manager plugin. Defaults to "1.0.0".
+-   **`mcpClients`**: (required, object) An object where each key is a client-side name (namespace) for an MCP server, and the value is the configuration for that server.
```

Comment:
This should be `mcpServers` (you're specifying the list of servers to connect to, also that's consistent with how this is configured in other tools).